devops/scripts/benchmarks/utils/unitrace.py

```diff
@@ -0,0 +1,186 @@
+
+        os.makedirs(bench_dir, exist_ok=True)
+
+        unitrace_output = os.path.join(
+            bench_dir, f"{self.timestamp}_{options.save_name}.out"
+        )
+
+        if extra_unitrace_opt is None:
+            extra_unitrace_opt = []
+
+        unitrace_command = (
+            [
+                str(unitrace_bin),
+                "--call-logging",
+                "--host-timing",
+                "--device-timing",
+                "--chrome-sycl-logging",
+                "--chrome-call-logging",
+                "--chrome-kernel-logging",
+                "--output",
+                unitrace_output,
+            ]
+            + extra_unitrace_opt
+            + command
+        )
+        log.debug(f"Unitrace cmd: {' '.join(unitrace_command)}")
+
+        return unitrace_output, unitrace_command
+
+    def handle_output(self, unitrace_output: str):
+
+        # Handle {name}.{pid}.json files in cwd: move and rename to {self.name()}_{timestamp}.{pid}.json
+        pid_json_files = []
+        pid = ""
+        for f in os.listdir(options.benchmark_cwd):
+            parts = f.split(".")
+            l = len(parts)
+            if len(parts) >= 3 and parts[l - 1] == "json" and parts[l - 2].isdigit():
+                pid_json_files.append(f)
+                pid = parts[l - 2]
+
+        if len(pid_json_files) == 0:
+            raise FileNotFoundError(
+                f"No .pid.json files found in {options.benchmark_cwd}."
+            )
+        elif len(pid_json_files) > 1:
+            # If there are multiple .pid.json files due to previous failures, keep only the most recent one
+            pid_json_files.sort(
+                key=lambda f: os.path.getmtime(os.path.join(options.benchmark_cwd, f))
+            )
+            for f in pid_json_files[:-1]:
+                os.remove(os.path.join(options.benchmark_cwd, f))
+            pid_json_files = [pid_json_files[-1]]
+
+        dst = (
+            unitrace_output[:-4] + f".{pid}.json"
+            if unitrace_output.endswith(".out")
+            else unitrace_output + f".{pid}.json"
+        )
+
+        shutil.move(os.path.join(options.benchmark_cwd, pid_json_files[0]), dst)
+        log.debug(f"Moved {pid_json_files[0]} to {dst}")
+
+        # Prune old unitrace directories
+        self._prune_unitrace_dirs(os.path.dirname(unitrace_output))
+
+
+_unitrace_instance = None
+
+
+def create_unitrace(inclusive: bool) -> None:
+    global _unitrace_instance
+    if _unitrace_instance is None:
+        try:
+            _unitrace_instance = Unitrace(inclusive)
+        except Exception as e:
+            log.error(f"Failed to build Unitrace: {e}")
+            _unitrace_instance = None
+        if _unitrace_instance is not None:
+            log.info("Unitrace instance created successfully.")
+    else:
+        raise ValueError("Unitrace instance already created")
+
+
+def get_unitrace() -> Unitrace | None:
+    return _unitrace_instance
```

Comment:
Use the same singleton pattern as other utils:
https://github.com/intel/llvm/blob/sycl/devops/scripts/benchmarks/utils/compute_runtime.py#L219-L222